botocore/client.py

```diff
@@ -208,6 +209,10 @@ def _get_client_args(self, service_model, region_name, is_secure,
         self._inject_s3_configuration(
             config_kwargs, scoped_config, client_config)
         self._conditionally_unregister_fix_s3_host(endpoint_url, event_emitter)
+        if service_name == 's3':
+            # We want to make double sure that we don't create this object
```

Comment:
I agree that it's more readable if we don't have the conditional.  I would actually prefer to remove as much service specific logic as possible in the client creator (I already think it's too much).